cmd/evict/evict.go

```diff
@@ -0,0 +1,193 @@
+	c := &evictCommand{
+		config: config,
+		client: client,
+	}
+
+	command := &cobra.Command{
+		Use:   "evict",
+		Short: "Evict removes a volume's replica from a node",
+		Example: `
+		$ storageos evict my-namespace my-volume my-node
+		$ storageos evict my-namespace my-volume my-node --cordon=false`,
+		Args: argwrappers.WrapInvalidArgsError(func(_ *cobra.Command, args []string) error {
+			if len(args) < 3 {
+				return errors.New("received too few args")
+			}
+			if len(args) > 3 {
+				return fmt.Errorf("received more args than expected: %v", args)
+			}
+			return nil
+		}),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			run := runwrappers.Chain(
+				runwrappers.RunWithTimeout(config),
+				runwrappers.EnsureNamespaceSetWhenUseIDs(config),
+				runwrappers.AuthenticateClient(config, client),
+			)(c.runWithCtx)
+
+			return run(context.Background(), cmd, args)
+		},
+		SilenceUsage: true,
+	}
+	command.Flags().BoolVar(&c.cordonTargetNode, "cordon", true, "cordon the target node before eviction")
+
+	return command
+}
+
+func (c *evictCommand) runWithCtx(ctx context.Context, cmd *cobra.Command, args []string) error {
+	useIDs, err := c.config.UseIDs()
+	if err != nil {
+		return err
+	}
+
+	nsID := id.Namespace(args[0])
+	vID := id.Volume(args[1])
+	targetNode := id.Node(args[2])
+
+	if !useIDs {
+
+		nn, err := c.client.GetNamespaceByName(ctx, string(nsID))
+		if err != nil {
+			return err
+		}
+		nsID = nn.ID
+
+		n, err := c.client.GetNodeByName(ctx, string(targetNode))
+		if err != nil {
+			return err
+		}
+		targetNode = n.ID
+
+		v, err := c.client.GetVolumeByName(ctx, nsID, string(vID))
+		if err != nil {
+			return err
+		}
+		vID = v.ID
+
+	}
+
+	vol, err := c.client.GetVolume(ctx, nsID, vID)
+	if err != nil {
+		return err
+	}
+
+	// Do some basic sanity checking to ensure we're not about to destroy a volume
+	if len(vol.Replicas) < 1 {
+		return fmt.Errorf("cannot evict volume if it has no replicas - data would be lost")
+	}
+
+	if vol.Master.Health != health.MasterOnline {
+		return fmt.Errorf("cannot evict volume - master unhealthy: %v", vol.Master.Health.String())
+	}
+	for _, replica := range vol.Replicas {
+		if replica.Health != health.ReplicaReady || replica.Promotable == false {
+			return fmt.Errorf("cannot evict volume - replica unhealthy, id: %v health: %v, promotable: %v", replica.ID, replica.Health, replica.Promotable)
+		}
+	}
+
+	// find the target deployment
+	var deploymentToRemove id.Deployment
+	if vol.Master.Node == targetNode {
+		return fmt.Errorf("cannot evict volume - master deployment cannot be evicted")
+	} else {
+		for _, replica := range vol.Replicas {
+			if replica.Node == targetNode {
+				deploymentToRemove = replica.ID
+				break
+			}
+		}
+	}
+
+	if deploymentToRemove == "" {
+		return errors.New("failed to find deployment on target node for volume")
+	}
+
+	if c.cordonTargetNode {
+		fmt.Println("cordining node")
```

Comment:
Are these debug statements?
Also, typo on the word "cordon"

Reply:
Nope, they're meant for user output, I didn't want to setup a logger for this relatively simple command (most over commands also use `fmt`.